### ISSUE TITLE
Change Request: Allow the name property and the ignores property to be used together.

### ISSUE THREAD

**Issue by Karibash** [eslint/eslint#18723]
### ESLint version

v9.8.0

### What problem do you want to solve?

If the name property and the ignores property are used together, the ignores property will not be applied globally.
Since the name property is used to identify the configuration object, it should be able to be used in conjunction with the ignores property.
```js
export default [
  {
    name: 'ignores',
    ignores: [
      '**/dist',
    ],
  },
  // other rules
];
```

Currently, if you want to globally apply the ignores property, you must use the ignore property on its own, as shown below.
```js
export default [
  {
    ignores: [
      '**/dist',
    ],
  },
  // other rules
];
```

### What do you think is the correct solution?

It is expected that the ignores property will be applied globally, even when the name and ignores properties are used together, as shown below.
```js
export default [
  {
    name: 'ignores',
    ignores: [
      '**/dist',
    ],
  },
  // other rules
];
```

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by blipk** [comment]
This is definately needed as I can't see another way to ignore a directory in the new flat config format, but I've set up all my rules etc in there which makes the ignores key non-global which is kind of useless and seems like very strange behaviour.

See: https://github.com/eslint/eslint/discussions/17429

**Comment by mdjermanovic** [comment]
> If the name property and the ignores property are used together, the ignores property will not be applied globally. Since the name property is used to identify the configuration object, it should be able to be used in conjunction with the ignores property.
> 
> ```js
> export default [
>   {
>     name: 'ignores',
>     ignores: [
>       '**/dist',
>     ],
>   },
>   // other rules
> ];
> ```

`ignores` are applied globally if `name` is the only other key in the config object. Can you provide a reproducible example where it doesn't work that way? Because that would be a bug.

**Comment by mdjermanovic** [comment]
@blipk you can ignore directories by specifying `ignores` in a config object that has no other keys (or the only other key is `name`).

https://eslint.org/docs/latest/use/configure/ignore

**Comment by blipk** [comment]
@mdjermanovic 

So I'm not allowed to configure anything else if I wan't global ignores?

That seems incredibly strange and useless.

**Comment by Karibash** [comment]
@mdjermanovic 
Ah... I'm sorry, it wasn't a bug. The issue was that I mistakenly ran it on v8 instead of v9.
It seems that the name property has also become usable with the PR below, but is it possible to backport this to v8?
There are still many plugins that do not yet support v9, so it would be great if we could use both name and ignores in v8 as well.
https://github.com/eslint/eslint/pull/18334/files#diff-891c5261db14a885f6a80ecc03f7b50f01939c66a798c14b1292621def227e82R151

**Comment by nzakas** [comment]
> It seems that the name property has also become usable with the PR below, but is it possible to backport this to v8?

I think this is something worth considering for compatibility reasons. We do need to complete some [infrastructure work](https://github.com/eslint/eslint/issues/18691) to allow us to backport, though.

@eslint/eslint-tsc thoughts?

**Comment by fasttime** [comment]
`@humanwhocodes/config-array` already includes the required fix: https://github.com/humanwhocodes/config-array/commit/3dabb4db2072a86d9b567205626ba0ed537bea2f. I think we could backport the changes to `eslint/eslint` once we have the infrastructure set.

**Comment by nzakas** [comment]
Ah weird, v8.57.0 is at `@humanwhocodes/config-array@0.11.14`. Looks like we never upgraded to include the fix.

But that does mean that fresh installs of v8.57.0 will get the update, at least.

**Comment by fasttime** [comment]
> Ah weird, v8.57.0 is at `@humanwhocodes/config-array@0.11.14`. Looks like we never upgraded to include the fix.
> 
> But that does mean that fresh installs of v8.57.0 will get the update, at least.

I think that a caret `^` before a version spec that starts with 0 only allows patch updates (see [here](https://nodesource.com/blog/semver-tilde-and-caret#majorzeroandminor10yz0yz0y10) for example). So, in our case `@humanwhocodes/config-array@0.11.14` will only match package versions of `0.11.x` where `x` is 14 or higher.

Anyway, I tried explicitly updating the `v8.x` branch to use `@humanwhocodes/config-array@0.13.0` (the latest version) and all tests were still passing, so I guess that should work at least to backport the fix. I also tried switching to `@eslint/config-array@0.17.1` but had no luck with the unit tests, with some tests failing because [`ConfigArray.isExplicitMatch()` has been removed](https://github.com/eslint/rewrite/commit/10d8200fbc99ae4496fc308af82930bb41aeacfa), or because of other changes.

**Comment by nzakas** [comment]
> I think that a caret `^` before a version spec that starts with 0 only allows patch updates (see [here](https://nodesource.com/blog/semver-tilde-and-caret#majorzeroandminor10yz0yz0y10) for example). So, in our case `@humanwhocodes/config-array@0.11.14` will only match package versions of `0.11.x` where `x` is 14 or higher.

Ah right. Always forget how those magic symbols work.

**Comment by mdjermanovic** [comment]
> I think this is something worth considering for compatibility reasons. We do need to complete some [infrastructure work](https://github.com/eslint/eslint/issues/18691) to allow us to backport, though.

Makes sense to me. I could work on this after 20th Aug.

**Comment by ReenigneArcher** [comment]
I'm new to eslint, but this is pretty weird... and not clear at all in the documentation at (https://eslint.org/docs/latest/use/configure/ignore).

doesn't work
```js
import globals from "globals";
import pluginJs from "@eslint/js";

export default [
  pluginJs.configs.recommended,
  {
    languageOptions: {
      globals: {
        ...globals.browser,
        ...globals.node,
        "require": "readonly",
      },
    },
    ignores: [
        "coverage/**",
        "dist/**",
    ],
    plugins: {
      jest: {
        extends: ["eslint:recommended"],
        rules: {
          "jest/no-disabled-tests": "error",
          "jest/no-focused-tests": "error",
          "jest/no-identical-title": "error",
          "jest/prefer-to-have-length": "error",
          "jest/valid-expect": "error",
        },
      },
    },
  },
];
```

works
```js
import globals from "globals";
import pluginJs from "@eslint/js";

export default [
  pluginJs.configs.recommended,
  {
    ignores: [
        "coverage/**",
        "dist/**",
    ],
  },
  {
    languageOptions: {
      globals: {
        ...globals.browser,
        ...globals.node,
        "require": "readonly",
      },
    },
    plugins: {
      jest: {
        extends: ["eslint:recommended"],
        rules: {
          "jest/no-disabled-tests": "error",
          "jest/no-focused-tests": "error",
          "jest/no-identical-title": "error",
          "jest/prefer-to-have-length": "error",
          "jest/valid-expect": "error",
        },
      },
    },
  },
];
```

**Comment by nzakas** [comment]
@ReenigneArcher this isn't the place to leave feedback. Please open a separate discussion if you need help.

**Comment by ReenigneArcher** [comment]
I didn't ask for help. I just posted examples to help others facing this.

**Comment by mdjermanovic** [comment]
Closing in favor of https://github.com/eslint/eslint/issues/18874.
